iOS/DuckDuckGo/AIChat/InputBox/SwitchBar/Transition/OmniBarEditingStateTransition.swift

```diff
@@ -72,25 +94,27 @@ class OmniBarEditingStateTransition: NSObject, UIViewControllerAnimatedTransitio
 
         containerView.addSubview(toVC.view)
 
-        let yOffset = toVC.switchBarVC.textEntryViewController.view.frame.minY * yOffsetFactor
+        let switchBarTextViewMinY = toVC.switchBarVC.textEntryViewController.view.frame.minY
+        let offsets = calculateOffsets(switchBarTextViewMinY: switchBarTextViewMinY)
 
-        toVC.view.frame = containerView.bounds.offsetBy(dx: 0, dy: -yOffset)
+        toVC.view.layer.sublayerTransform = CATransform3DMakeTranslation(0, -offsets.switcherYOffset, 0)
```

Comment:
Turns out this transform is safer than manipulating frames and allows to keep the background intact, e.g. when content moves out of bounds.